app/src/main/java/de/grobox/transportr/trips/detail/TripDetailFragment.kt

```diff
@@ -0,0 +1,203 @@
+import androidx.recyclerview.widget.RecyclerView
+import androidx.appcompat.widget.Toolbar
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+
+import javax.inject.Inject
+
+import de.grobox.transportr.R
+import de.grobox.transportr.TransportrFragment
+import de.grobox.transportr.trips.detail.TripUtils.getStandardFare
+import de.grobox.transportr.trips.detail.TripUtils.hasFare
+import de.grobox.transportr.trips.detail.TripDetailViewModel.SheetState
+import de.grobox.transportr.utils.TransportrUtils.getColorFromAttr
+import de.schildbach.pte.dto.Trip
+
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import androidx.core.content.ContextCompat.getColor
+import androidx.lifecycle.Observer
+import de.grobox.transportr.trips.detail.TripDetailViewModel.SheetState.*
+import de.grobox.transportr.trips.detail.TripUtils.intoCalendar
+import de.grobox.transportr.trips.detail.TripUtils.share
+import de.grobox.transportr.utils.DateUtils.*
+
+class TripDetailFragment : TransportrFragment(), Toolbar.OnMenuItemClickListener {
+
+    companion object {
+        val TAG = TripDetailFragment::class.java.simpleName
+    }
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    private lateinit var viewModel: TripDetailViewModel
+    private lateinit var toolbar: Toolbar
+    private lateinit var list: RecyclerView
+    private lateinit var bottomBar: View
+    private lateinit var fromTime: TextView
+    private lateinit var from: TextView
+    private lateinit var toTime: TextView
+    private lateinit var to: TextView
+    private lateinit var duration: TextView
+    private lateinit var price: TextView
+    private lateinit var topBar: View
+    private lateinit var fromTimeRel: TextView
+    private lateinit var durationTop: TextView
+    private lateinit var priceTop: TextView
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val v = inflater.inflate(R.layout.fragment_trip_detail, container, false)
+        setHasOptionsMenu(true)
+        component.inject(this)
+
+        toolbar = v.findViewById(R.id.toolbar)
+        list = v.findViewById(R.id.list)
+        bottomBar = v.findViewById(R.id.bottomBar)
+        fromTime = bottomBar.findViewById(R.id.fromTime)
+        from = bottomBar.findViewById(R.id.from)
+        toTime = bottomBar.findViewById(R.id.toTime)
+        to = bottomBar.findViewById(R.id.to)
+        duration = bottomBar.findViewById(R.id.duration)
+        price = bottomBar.findViewById(R.id.price)
+        topBar = v.findViewById(R.id.topBar)
+        fromTimeRel = topBar.findViewById(R.id.fromTimeRel)
+        durationTop = topBar.findViewById(R.id.durationTop)
+        priceTop = topBar.findViewById(R.id.priceTop)
```

Comment:
You should be able to remove those and use the kotlin databinding instead

Reply:
I'm not sure if I understand your proposal: Do you mean by just removing the `.findViewById(R.id.` part or is it even possible to remove all of these lines and initialize them somehow right away starting from line 68?

Reply:
You should be able to use Kotlin ViewBinding https://kotlinlang.org/docs/tutorials/android-plugin.html#view-binding

Reply:
Okay, nice hint, this is really useful!